index.d.ts

```diff
@@ -95,4 +95,122 @@ declare module 'react-spring' {
       JSX.IntrinsicElements[Tag]
     >
   };
+
+  type TransitionKeyProps = string | number;
+  type TransitionItemProps = string | number | object;
+
+  type TransitionProps<S extends object, DS extends object = {}> = {
+    /**
+     * Will skip rendering the component if true and write to the dom directly
+     * @default false
+     */
+    native?: boolean;
+    /**
+     * Spring config ({ tension, friction })
+     * @default config.default
+     */
+    config?: SpringConfig | ((key: string) => SpringConfig);
+    /**
+     * Base styles
+     * @default {}
+     */
+    from?: DS;
+    /**
+     * Animated styles when the component is mounted
+     * @default {}
+     */
+    enter?: DS;
+    /**
+     * Unmount styles
+     * @default {}
+     */
+    leave?: DS;
+
+    update?: DS;
+    /**
+     * A collection of unique keys that must match with the childrens order
+     * Can be omitted if children/render aren't an array
+     * Can be a function, which then acts as a key-accessor which is useful when you use the items prop
+     * @default {}
+     */
+    keys: ((params: any) => TransitionKeyProps) | Array<TransitionKeyProps> | TransitionKeyProps;
+    /**
+     * Optional. Let items refer to the actual data and from/enter/leaver/update can return per-object styles
+     * @default {}
+     */
+    items?: Array<TransitionItemProps> | TransitionItemProps;
+
+    children?: (
+      params: DS & S,
+    ) => ReactNode | Array<(params: DS & S) => ReactNode>;
+
+    render?: (
+      params: DS & S,
+    ) => ReactNode | Array<(params: DS & S) => ReactNode>;
+  };
+
+  export class Transition<
+    S extends object,
+    DS extends object,
```

Comment:
I'm getting TS1009 no trailing spaces allowed on the latest version of TS, could you please remove them?